csharp/src/Ice/Ice1Definitions.cs

```diff
@@ -144,16 +144,17 @@ internal static DispatchException ReadIce1SystemException(this InputStream istr,
 
                     if (replyStatus == ReplyStatus.OperationNotExistException)
                     {
-                        systemException = new OperationNotExistException(identity, facet, operation);
+                        systemException = new OperationNotExistException();
                     }
                     else
                     {
-                        systemException = new ObjectNotExistException(identity, facet, operation);
+                        systemException = new ObjectNotExistException();
                     }
+                    systemException.Origin = new RemoteExceptionOrigin(identity, facet, operation);
```

Comment:
I would use the ctor that takes a RemoteExceptionOrigin paramater instead of setting it later.